Teacher.java

```diff
@@ -0,0 +1,16 @@
+/*
+Create class Teacher with the following:
+Inherits from Person.
+ */
+public class Teacher extends Person{
+ String specialization;
+    //Constructor extends parent's constructor by adding specialization and a parameter for it.
+    public Teacher(String name, int age, boolean parent_permission) {
+        super(name, age, parent_permission);
+        this.specialization=specialization;
```

Comment:
Specialization not defined  in public Teacher(String name, int age, boolean parent_permission)